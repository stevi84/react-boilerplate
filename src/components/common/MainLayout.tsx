import React from 'react';
import { useNotifier } from '../../hooks/UseNotifier';
import { Header } from './Header';

export const MainLayout = (props: React.PropsWithChildren) => {
  useNotifier();

  return (
    <>
      <Header />
      {props.children}
    </>
  );
};
