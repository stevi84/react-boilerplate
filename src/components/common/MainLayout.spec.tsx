import { describe, it, expect, vi } from 'vitest';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import { MainLayout } from './MainLayout';

vi.mock('../../hooks/UseNotifier', () => ({ useNotifier: vi.fn() }));

describe('MainLayout', () => {
  it('should equal saved snapshot', () => {
    const renderer: ShallowRenderer = createRenderer();
    renderer.render(<MainLayout />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
