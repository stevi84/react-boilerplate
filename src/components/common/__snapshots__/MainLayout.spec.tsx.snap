// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainLayout > should equal saved snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-5czdux-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-95f9g5-MuiToolbar-root"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-1by7r3h-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
          />
        </svg>
      </button>
      <div
        class="MuiTypography-root MuiTypography-h6 css-173147z-MuiTypography-root"
      >
        React Boilerplate
      </div>
      <div
        class="MuiTypography-root MuiTypography-body1 css-kcbyr-MuiTypography-root"
      >
        Environment: test
      </div>
      <div
        class="MuiBox-root css-ece9u5"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-1b5lkxe-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-18pjc51-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
          >
            language
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1cprn2g-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-controls=":r0:"
              aria-expanded="false"
              aria-haspopup="listbox"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-15k6ek6-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              role="combobox"
              tabindex="0"
            >
              de
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="de"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1u6jos5-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-kg5swy"
              >
                <span>
                  language
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-95f9g5-MuiToolbar-root"
  />
</DocumentFragment>
`;
